### PR TITLE
chore: refactor common code across lambda functions and fixes incorrect typing

### PR DESCRIPTION
**Pull request by as1729**
This PR was created to fix two things:

1. Typing errors within the python code related to `TemporaryFileWrapper`.
2. Issue with the usage of `json.dump`. Unclear why this wasn't generating test failures but the function call was raising a type-issue for a missing argument.
3. Consolidates the generation for s3 file keys for output files and stores it within `treasury_generation_common.py`. While this doesn't reduce the lines of code- I'm hoping that consolidating this reduces the chance of having a malformed object key anywhere else in the code as it appears to be recreated in multiple places.

### CODE REVIEW

File: python/src/functions/generate_treasury_report.py [hunk 1]
```diff
@@ -215,16 +223,24 @@ def process_event(payload: ProjectLambdaPayload, logger):
...
+                key=get_generated_output_file_key(
+                    file_type=OutputFileType.CSV,
+                    project=project_use_code,
+                    organization=organization,
+                ),
                 file=csv_file,
             )
         # Store project_id_agency_id to row number in a json file
         with tempfile.NamedTemporaryFile("w") as json_file:
-            json_file = json.dump(project_agency_id_to_row_map)
```
Comment (as1729) [inline python/src/functions/generate_treasury_report.py:223]: @vshia would be great to get your input here. I updated this based on typing errors I noticed. However I wasn't sure if this breaks any of the existing functionality.

File: python/src/functions/generate_treasury_report.py [hunk 2]
```diff
@@ -177,6 +178,10 @@ def process_event(payload: ProjectLambdaPayload, logger):
             with tempfile.NamedTemporaryFile() as file:
                 # Download projects from S3
                 created_at = file_info.createdAt
+
+                # TODO: Add a verification to ensure that the objectKey here is formatted correctly.
+                # Path must be: uploads/{organization_id}/{agency_id}/{reporting_period_id}/{upload_id}/{filename}.xlsm
```
Comment (laurelkorwin) [inline python/src/functions/generate_treasury_report.py:183]: You could potentially borrow logic here from `matchRegex` in `processValidationJson`, which has the same key structure

File: python/src/lib/treasury_generation_common.py [hunk 3]
```diff
@@ -22,6 +23,17 @@ class UserObj(BaseModel):
...
     email: str
 
+class OutputFileType(Enum):
+    XLSX = "xlsx"
+    CSV = "csv"
+    JSON = "json"
+
+
+def get_generated_output_file_key(
+    file_type: OutputFileType, project: str, organization: OrganizationObj
```
Comment (laurelkorwin) [inline python/src/lib/treasury_generation_common.py:33]: any appetite for making `project` values an Enum too? I know we're grabbing from the dictionary below so perhaps not strictly necessary but could make the flow here more legible?

**Review by laurelkorwin — APPROVED** [review]

File: python/src/lib/treasury_generation_common.py (hunk 3 above)
Comment (as1729) [reply python/src/lib/treasury_generation_common.py:33]: 💯  - I'll create a separate PR with this updated

File: python/src/functions/generate_treasury_report.py (hunk 2 above)
Comment (as1729) [reply python/src/functions/generate_treasury_report.py:183]: 💯 - will address in a separate PR

**Pull request merged**